quicfire_tools/inputs.py

```diff
@@ -621,28 +636,184 @@ def set_output_files(
+            )
+        else:
+            self.windsensors.update_sensor(
+                sensor_name=update,
+                wind_times=wind_times,
+                wind_speeds=wind_speeds,
+                wind_directions=wind_directions,
+                sensor_height=sensor_height,
+                x_location=x_location,
+                y_location=y_location,
+            )
+
+    def new_wind_sensor_from_csv(
+        self, directory: str | Path, filename: str, update: str = None
+    ):
+        """
+        Adds a wind sensor or updates an existing one from a csv file.
+
+        Parameters
+        ----------
+        directory : str | Path
+            Directory containing the csv to read
+        filename : str
+            Name of the csv file
+        update: str
+            Name of the wind sensor, e.g. "sensor1". Sensor must already exist in windsensors.
+            Omit or use update = None to add a new sensor.
+
+        Columns
+        -------
+        wind_times : int >= 0
+            Time in seconds that each windshift occurs from the start of the simulation.
+            First entry must be 0.
+        wind_speeds : float > 0
+            Wind speed of each windshift in m/s.
+        wind_directions : 0 <= int < 360
+            Wind direction of each windshift in degrees.
+        sensor_height : float > 0
+            Height of wind sensor. All values in column must be the same
+        x_location : int >= 0
+            Location of the wind sensor in the x-direction. All values in column must be the same
+        y_location : int >= 0
+            Location of the wind sensor in the y-direction. All values in column must be the same
+
+        Examples
+        --------
+        >>> from quicfire_tools import SimulationInputs
+        >>> sim_inputs = SimulationInputs.create_simulation(nx=100, ny=100, fire_nz=26, wind_speed=1.8, wind_direction=90, simulation_time=600)
+        >>> sim_inputs.new_wind_sensor_from_csv("path/to/directory","filename",update='sensor1')
+        """
+        if isinstance(directory, str):
+            directory = Path(directory)
+        filepath = directory / filename
+        df = pd.read_csv(filepath)
+        validate = DataFrameSchema(
+            {
+                "wind_times": Column(int, checks=Check.ge(0)),
+                "wind_speeds": Column(float, checks=Check.gt(0)),  # can windspeed be 0?
+                "wind_directions": Column(int, checks=Check.lt(360)),
+                "sensor_height": Column(
+                    float,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
+                "x_location": Column(
+                    int,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
+                "y_location": Column(
+                    int,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
```

Comment:
What do you think of dropping these from the csv/DataFrame and instead include them as just an input to the new_sensor_from_csv method? It seems odd to me to require three columns that are all the same value for every row.